BlueprintUICommonControls/Sources/AccessibilityFocus.swift

```diff
@@ -0,0 +1,117 @@
+    public var wrapping: Element
+
+    /// The type of accessibility notification that will be triggered.
+    public var notificationType: UIAccessibility.Notification
+
+    /// A object that can be held on to by the caller to manually trigger a focus.
+    public var trigger: Trigger
+
+    // MARK: Element
+
+    public var content: ElementContent {
+        ElementContent(child: wrapping)
+    }
+
+    public func backingViewDescription(with context: ViewDescriptionContext) -> ViewDescription? {
+        View.describe { config in
+            config[\.isAccessibilityElement] = false
+            config.apply { view in
+                view.apply(model: self)
+            }
+        }
+    }
+}
+
+// MARK: Backing view
+
+extension AccessibilityFocus {
+    private final class View: UIView {
+
+        func apply(model: AccessibilityFocus) {
+            model.trigger.action = { [weak self] in
+                guard let self = self else { return }
+                UIAccessibility.post(notification: model.notificationType, argument: self)
+            }
+
+            /// We need to wait a moment before firing the focus, to ensure that
+            /// we are not competing with UIAccessibility for the focus.
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
+                model.trigger.fireImmediately()
+            }
+        }
+    }
+}
+
+// MARK: Trigger
+
+extension AccessibilityFocus {
+
+    /// A trigger that can be used to manually fire an accessibility focus.
+    public class Trigger {
+
+        public let firesImmediately: Bool
+
+        /// Creates a new trigger for the purpose of changing accessibility focus.
+        /// - Parameters:
+        ///   - firesImmediately: If `true`, the trigger will fire immediately when the element appears.
+        public init(firesImmediately: Bool = false) {
+            self.firesImmediately = firesImmediately
+        }
+
+        fileprivate var action: (() -> Void)?
+        private var hasFired = false
+
+        fileprivate func fireImmediately() {
+            if firesImmediately && !hasFired {
+                fire()
+            }
+        }
+
+        /// Manually fire the trigger
+        public func fire() {
+            guard let action else { return }
+
+            action()
+            hasFired = true
+        }
+    }
+}
+
+extension Element {
+
+    /// Enables VoiceOver focus to jump to the wrapped element via the trigger.
+    /// - Parameters:
+    ///   - notificationType: Type of accessibility notification to trigger. Defaults to `.layoutChanged`. Limited to `.layoutChanged` or `.screenChanged`.
+    ///   - trigger: A reference-type trigger object that can be used to trigger accessibility focus immediately on appearance or at a later point via the `fire()` function.
+    public func accessibilityFocus(
+        notificationType: UIAccessibility.Notification = .layoutChanged,
+        trigger: AccessibilityFocus.Trigger
```

Comment:
just a little bikeshedding:
```suggestion
        on trigger: AccessibilityFocus.Trigger
```